test/unit-test/fleet_provisioning_utest.c

```diff
@@ -156,6 +156,44 @@ int suiteTearDown( int numFailures )
 }
 /*-----------------------------------------------------------*/
 
+/* Prototypes for test functions. */
+void test_RegisterThing_MacrosString( void );
```

Comment:
Is the addition of prototypes of the function only for satisfying clang? 
We don;t necessarily need to have the test comply with clang as this isn't adding much value. Only the library code itself can be kept compliant with clang

Reply:
Its for fixing a warning. Gcc might have this warning and we may not have it enabled. Its a small fix to have all the source files clean of warnings though, so it is likely worth it.